src/components/tl/ucp/tl_ucp_service_coll.c

```diff
@@ -34,15 +35,32 @@ ucc_status_t ucc_tl_ucp_service_allreduce(ucc_base_team_t *team, void *sbuf,
             .mem_type = UCC_MEMORY_TYPE_HOST
         }
     };
-    ucc_coll_task_init(&task->super);
+    status = ucc_coll_task_init(&task->super);
+    if (status != UCC_OK) {
+        goto free_task;
+    }
     task->subset = subset;
     task->team = tl_team;
     task->tag  = UCC_TL_UCP_SERVICE_TAG;
     task->n_polls = 10; // TODO need a var ?
     task->super.progress = ucc_tl_ucp_allreduce_knomial_progress;
     memcpy(&task->args, &args, sizeof(ucc_coll_args_t));
     *task_p = &task->super;
-    return ucc_tl_ucp_allreduce_knomial_start(&task->super);
+    status = ucc_tl_ucp_allreduce_init(task);
```

Comment:
this looks strange. we use knomial start below but for init we are now using old wrapper (ucc_tl_ucp_allreduce_init) which is not necessarily connected to knomial alg. why can't we call ucc_tl_ucp_allreduce_knomial_init_common here?

Reply:
sure, can use ucc_tl_ucp_allreduce_knomial_init_common